src/commands/layout/pull.ts

```diff
@@ -0,0 +1,196 @@
+    );
+
+    await EmailLayout.writeEmailLayoutDirFromData(dirContext, resp.data);
+
+    const action = dirContext.exists ? "updated" : "created";
+    this.log(
+      `‣ Successfully ${action} \`${dirContext.key}\` at ${dirContext.abspath}`,
+    );
+  }
+
+  // Pull all email layouts
+  async pullAllEmailLayouts(): Promise<void> {
+    const { flags } = this.props;
+
+    const defaultToCwd = { abspath: this.runContext.cwd, exists: true };
+    const targetDirCtx = flags["layouts-dir"] || defaultToCwd;
+
+    const prompt = targetDirCtx.exists
+      ? `Pull latest layouts into ${targetDirCtx.abspath}?\n  This will overwrite the contents of this directory.`
+      : `Create a new layouts directory at ${targetDirCtx.abspath}?`;
+
+    const input = flags.force || (await promptToConfirm(prompt));
+    if (!input) return;
+
+    spinner.start(`‣ Loading`);
+
+    const emailLayouts = await this.listAllEmailLayouts();
+
+    await EmailLayout.writeEmailLayoutIndexDir(targetDirCtx, emailLayouts);
+    spinner.stop();
+
+    const action = targetDirCtx.exists ? "updated" : "created";
+    this.log(
+      `‣ Successfully ${action} the layouts directory at ${targetDirCtx.abspath}`,
+    );
+  }
+
+  async listAllEmailLayouts(
+    pageParams: Partial<PageInfo> = {},
+    emailLayoutsFetchedSoFar: EmailLayout.EmailLayoutData<WithAnnotation>[] = [],
+  ): Promise<EmailLayout.EmailLayoutData<WithAnnotation>[]> {
+    const props = merge(this.props, {
+      flags: {
+        ...pageParams,
+        annotate: true,
+        limit: MAX_PAGINATION_LIMIT,
+      },
+    });
+
+    const resp = await this.apiV1.listEmailLayouts<WithAnnotation>(props);
+    if (!isSuccessResp(resp)) {
+      const message = formatErrorRespMessage(resp);
+      this.error(new ApiError(message));
+    }
+
+    const { entries, page_info: pageInfo } = resp.data;
+    const emailLayouts = [...emailLayoutsFetchedSoFar, ...entries];
+
+    return pageInfo.after
+      ? this.listAllEmailLayouts({ after: pageInfo.after }, emailLayouts)
+      : emailLayouts;
+  }
+
+  async getEmailLayoutDirContext(): Promise<EmailLayoutDirContext> {
+    const { emailLayoutKey } = this.props.args;
+    const { resourceDir, cwd: runCwd } = this.runContext;
+
+    // Inside an existing resource dir, use it if valid for the target email layout.
+    if (resourceDir) {
+      const target: ResourceTarget = {
+        commandId: BaseCommand.id,
+        type: "email_layout",
+        key: emailLayoutKey,
+      };
+
+      return ensureResourceDirForTarget(
+        resourceDir,
+        target,
+      ) as EmailLayoutDirContext;
+    }
+
+    // Not inside any existing email layout directory, which means either create a
+    // new email layout directory in the cwd, or update it if there is one already.
+    if (emailLayoutKey) {
+      const dirPath = path.resolve(runCwd, emailLayoutKey);
+      const exists = await EmailLayout.isEmailLayoutDir(
+        dirPath,
+        `${emailLayoutKey}.json`,
```

Comment:
I haven't looked at the PR in details, but just jumping in to share quick thoughts based on the questions above.

As far as the directory structure goes, I would say the workflows directory is the example we want to mirror for email layouts rather than the translations one. I view the translations dir as more of an exception to the rule. 

What I had in mind for email layouts is to structure the directory like:
```
layouts/
  default/
    layout.json
    html_layout.html
    text_layout.txt
  another_layout/
    layout.json
    html_layout.html
    text_layout.txt
...
```
Basically: all email layouts live in the `layouts` directory, and inside each layout gets its own dir named by its key. And inside you always expect to have a `layout.json` file, plus html and text layout files that are referenced by the `layout.json` file.

I expect this is the directory structure we will replicate for other resources including email layouts, unless there's a good reason/exception not to.

Translations are in that "exception" category, mainly because:
1. Translations do not have a "key" like workflows or layouts, but is identified by its locale + namepsace (like a composite key). But there is a natural grouping between them where all translations of the same locale belong to one group.
2. Unlike workflows or email layouts, translations do not really have any meta data around them, the **only** real data it contains is the translation content.
3. Most importantly, it's possible that some customers might be using a 3rd party (or open source) translations management service as a source of truth for all translations in their product, and only need to export and import a subset of translations into Knock for comms. In that scenario, we want to make that process as easy and seamless as possible to go between that service and Knock. Typically those services let you download translations in json or yaml file formats, containing translation mappings, named in the format of locale + namepsace, inside each locale directory. And that's the general structure of how our translations directory is structured as well. 

Let me know what you think, happy to chat this through live if helpful!

Reply:
OK got it! Thanks for the explanation on this!